simulator/src/main/java/com/github/benmanes/caffeine/cache/simulator/policy/AccessEvent.java

```diff
@@ -27,69 +27,122 @@
  * @author ben.manes@gmail.com (Ben Manes)
  */
 public class AccessEvent {
-  private final Long key;
-
-  private AccessEvent(long key) {
-    this.key = key;
-  }
-
-  /** Returns the key. */
-  public Long key() {
-    return key;
-  }
-
-  /** Returns the weight of the entry. */
-  public int weight() {
-    return 1;
-  }
-
-  @Override
-  public boolean equals(Object o) {
-    if (o == this) {
-      return true;
-    } else if (!(o instanceof AccessEvent)) {
-      return false;
+    private final Long key;
+
+    private AccessEvent(long key) {
+        this.key = key;
     }
-    AccessEvent event = (AccessEvent) o;
-    return Objects.equals(key(), event.key())
-        && Objects.equals(weight(), event.weight());
-  }
-
-  @Override
-  public int hashCode() {
-    return Objects.hash(key(), weight());
-  }
-
-  @Override
-  public String toString() {
-    return MoreObjects.toStringHelper(this)
-        .add("key", key())
-        .add("weight", weight())
-        .toString();
-  }
-
-  /** Returns an event for the given key. */
-  public static AccessEvent forKey(long key) {
-    return new AccessEvent(key);
-  }
-
-  /** Returns an event for the given key and weight. */
-  public static AccessEvent forKeyAndWeight(long key, int weight) {
-    return new WeightedAccessEvent(key, weight);
-  }
-
-  private static final class WeightedAccessEvent extends AccessEvent {
-    private final int weight;
-
-    WeightedAccessEvent(long key, int weight) {
-      super(key);
-      this.weight = weight;
-      checkArgument(weight >= 0);
+
+    /**
+     * Returns the key.
+     */
+    public long key() {
+        return key;
     }
 
-    @Override
+    /**
+     * Returns the weight of the entry.
+     */
     public int weight() {
-      return weight;
+        return 1;
+    }
+
+    /**
+     * Returns the miss penalty of the entry
+     */
+    public int missPenalty() {
+        return 1;
+    }
+
+    /**
+     * Returns the hit penalty of the entry
+     */
+    public int hitPenalty() {
+        return 0;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == this) {
+            return true;
+        } else if (!(o instanceof AccessEvent)) {
+            return false;
+        }
+        AccessEvent event = (AccessEvent) o;
+        return Objects.equals(key(), event.key())
```

Comment:
All of the new fields should be added to the equals/hashCode/toString

Reply:
Fixed in the latest push

Reply:
in order to pass the travis.sh analysis, I needed to override equals and hashCode inside the PenaltiesAccessEvent class.
when I just added the fields to the methods that are defined in AccessEvent the analysis failed due to:
EQ_DOESNT_OVERRIDE_EQUALS: Class doesn't override equals in superclass

if you have another idea how to overcome this warning I could make the changes, but right now that's the only way I was able to make this warning go away.

Reply:
Yes, please add it to the exclusions instead, which for Spotbugs is in `config/excludes.xml`:

https://github.com/ben-manes/caffeine/blob/aa0cdfad2e94e4a39432f7d27f43ff0a3e25755a/config/spotbugs/exclude.xml#L140-L143

Reply:
No problem, doing it now